# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/arm64/Lowering-arm64.h
# Commit: e2105a52bcdd
# Full Hash: e2105a52bcdd6229d2b8b5ee1a59bd8aee749821
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-10-02 04:10:47
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 6.4: Add tls dependency for WasmTruncateToInt64 and Int64ToFloatingPoint for arm. r=lth
#   
#   In this patch we add a tls dependency for the remaining nodes which use
#   BuiltinThunk to call c++ runtime. By ABI requirements WasmTlsReg should
#   be set.
# ==============================================================================

diff -r bba9e8259714 -r e2105a52bcdd js/src/jit/arm64/Lowering-arm64.h
--- a/js/src/jit/arm64/Lowering-arm64.h	Thu Oct 01 07:52:37 2020 +0000
+++ b/js/src/jit/arm64/Lowering-arm64.h	Thu Oct 01 07:52:46 2020 +0000
@@ -68,6 +68,8 @@
   void lowerForFPU(LInstructionHelper<1, 2, Temps>* ins, MDefinition* mir,
                    MDefinition* lhs, MDefinition* rhs);
 
+  void lowerBuiltinInt64ToFloatingPoint(MBuiltinInt64ToFloatingPoint* ins);
+  void lowerWasmBuiltinTruncateToInt64(MWasmBuiltinTruncateToInt64* ins);
   void lowerForBitAndAndBranch(LBitAndAndBranch* baab, MInstruction* mir,
                                MDefinition* lhs, MDefinition* rhs);
   void lowerTruncateDToInt32(MTruncateToInt32* ins);